train_model_lightning.py

```diff
@@ -0,0 +1,187 @@
+# General
+from datetime import datetime
+import json
+import os
+import signal
+import sys
+import yaml
+
+# Torch
+import torch
+from torchinfo import summary as model_summary
+
+# PyTorch Lightning
+from pytorch_lightning import Trainer
+from pytorch_lightning.plugins.environments import SLURMEnvironment
+from pytorch_lightning.callbacks import ModelCheckpoint
+from pytorch_lightning import loggers as pl_loggers
+
+# Hugging Face
+from transformers import set_seed
+
+# Other
+from pathlib import Path
+from tabulate import tabulate
+
+# Local
+from models import RetNetModel, TransformerModel
+from dataset import DataModule
+from utils import Struct
+
+# Allow torch to run float32 matrix multiplications in lower precision for
+# better performance while training if hardware is capable
+torch.backends.cuda.matmul.allow_tf32 = True
+
+class CustomCheckpoint(ModelCheckpoint):
```

Comment:
Can we rename this possibly to `CustomModelCheckpoint` or something similar to be more descriptive?

Reply:
changed